### ISSUE TITLE
Force reloading

### ISSUE THREAD

**Issue by julientregoat** [passcod/cargo-watch#123]
Hi!

I've been using `cargo-watch` since I started using Rust and I'm really really loving it. Thank you so much for all of your work so far.

Currently I'm building a library, and I'm also building a server that uses that library. I work on both simultaneously occasionally, and sometimes the library crashes the server, which I'm hot reloading with `cargo watch`. When I update code in the library, it doesn't cause the server to hot reload. This makes sense to me - a dependency, even local, changing shouldn't cause a sudden reload (in my opinion) unless I'm explicitly watching it. 

_However_, it'd be cool to reload the watched code at any point in time, by pressing a key like `r` or `<spacebar>` in the terminal or something. Feels like it's something that might be useful to others as well. Also, if you're interested, I'd love to contribute the code for this if you can point me in the right direction.

Thanks!

**Comment by passcod** [comment]
Yeah, this would be great. Would have to go in watchexec: https://github.com/watchexec/watchexec/issues/67

The way to go, I think, would be to start here: https://github.com/watchexec/watchexec/blob/master/src/run.rs#L93 and upgrade the channel event to support both filesystem events and keyboard events, then spawn another thread to wait on STDIN.

Also upgrade `on_manual` https://github.com/watchexec/watchexec/blob/master/src/run.rs#L193 to take a `restart: bool` so `rs<enter>` always restarts the process. Ideally I'd like to support both `entr`'s and `nodemon`'s inputs (`rs<enter>` = restart, `<space>` = go if stopped, `q` = stop gracefully).

Of course it would also need a flag to disable (and maybe customise the keybindings if ambitious! at least via the Rust API) the behaviour, for programs that want to do their own STDIN handling.

**Comment by julientregoat** [comment]
Okay, awesome, thank you for the guidance! I'll get started on this later this week/end.

**Comment by julientregoat** [comment]
Hey! I haven't forgotten about this; I'm about 35%, but got a bit slammed with stuff for a week so have to put it on hold for a little.

**Comment by passcod** [comment]
That's perfectly alright, thanks for the status update!

**Comment by TheZoq2** [comment]
Have you made any progress on this? I would love to see this functionality in `watchexec`

**Comment by julientregoat** [comment]
I got about halfway but then things got pretty busy this year and I haven't been able to tackle any of the OSS projects I've been wanting to. sorry bout the dealy! I still want this functionality as well and will try to finish it off in the next couple of months but someone's welcome to grab this if they want.

**Comment by TheZoq2** [comment]
Ah right, I might have a look at it if I get some time to spare

**Comment by passcod** [comment]
Closing in favour of upstream issue.
